src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/overview/MainAccordion.tsx

```diff
@@ -42,16 +43,28 @@ type Props = {
   updateEntityUI: (uIstate: { [key: string]: JSONValue }) => void;
 };
 
+enum AccordionSectionID {
+  FUNCTION = 'function',
+  SEQUENCE = 'sequence',
+  OTHER_DATA_SETS = 'other_data_sets'
+}
+
 // TODO: Remove me once instant download component is available
 const mockOnClick = noop;
 
 const MainAccordion = (props: Props) => {
-  if (!props.sidebarPayload) {
+  const { sidebarPayload, sidebarUIState } = props;
+
+  if (!sidebarPayload) {
     return null;
   }
-  const { gene } = props.sidebarPayload;
-  const expandedPanels = props.sidebarUIState?.mainAccordion
-    ?.expandedPanels as string[];
+
+  const { gene } = sidebarPayload;
+  const hasFunctionDescription = Boolean(gene.function?.description);
+
+  const expandedPanels =
+    (sidebarUIState?.mainAccordion?.expandedPanels as string[]) ||
+    (hasFunctionDescription ? [AccordionSectionID.FUNCTION] : []);
 
   const onChange = (expandedPanels: (string | number)[] = []) => {
```

Comment:
The types are all funny!

First, we say that accordion section id can only be one of these three values: 'function', 'sequence', or 'other_data_sets'.

Then (line 66) the type of `expandedPanels` is broadened to `string[]` (why? it should still be only one of the three string values).

Then, in the line I am leaving this comment at,  the type of expanded panels is broadened even further — to `(string | number)[]`.

This doesn't look right.

Reply:
> Then (line 66) the type of expandedPanels is broadened to string[] (why? it should still be only one of the three string values)

Replaced `string[]` with `AccordionSectionID[]`

>Then, in the line I am leaving this comment at, the type of expanded panels is broadened even further — to (string | number)[].

This is because using `AccordionSectionID[]` breaks the type as the accordion `onChange` function returns `(string | number)[]`.
I have now changed it to `AccordionSectionID[]` and the `onChange` callback to `onChange={(expandedPanels) => onChange(expandedPanels as AccordionSectionID[])}`. Would this work?

Reply:
> Would this work?

Yes, it'll work. It's a bit unfortunate that we had to add a new function for the sole purpose of narrowing down the types. It would probably be better to leave the onChange as is (`onChange = (expandedPanels: (string | number)[] = [])`), but then inside he body of the onChange do:

```
props.updateEntityUI({
      mainAccordion: {
        expandedPanels as AccordionSectionID[]
      }
    });
```

Reply:
That's done